### ISSUE TITLE
idea: second part of violations

### ISSUE THREAD

**Issue by Nimfadora** [checkstyle/checkstyle#4708]
sub-task of #4681

[report on violations](https://nimfadora.github.io/idea/suppressed_2_part.html)

violations names included:
```
AnonymousClassMethodCount:true Production:true
AnonymousInnerClass
AntResolveInspection
AssignmentToNull
AssignmentToStaticFieldFromInstanceMethod
ClassComplexity
ClassCoupling
ClassInheritanceDepth
CloneableImplementsClone
ConstantDeclaredInAbstractClass
DeprecatedIsStillUsed
Deprecation
FieldCanBeLocal
FieldCount
FinalClass
FinalMethod
MethodCount
NoopMethodInAbstractClass
PublicConstructor
SystemOutErr
UtilityClass
UtilityClassCanBeEnum
```

**Comment by Nimfadora** [comment]
SystemOutErr review [can be skipped](https://github.com/checkstyle/checkstyle/issues/4681#issuecomment-314614334)

**Comment by romani** [comment]
> AntResolveInspection

suppress "maven and ant are used in the same build, no options in inspection to adjust"

> AssignmentToNull

suppress: "we are legacy library with log history, to much items to fix, usage of Option is not do code any better in some cases, one day in future.... "

> AssignmentToStaticFieldFromInstanceMethod

try to fix.
case of `FileContentsHolder` suppress by javadoc annotation, this class will be removed soon.

> AnonymousClassMethodCount

make for test scope limit to be 3 . It should calm down.

> ClassInheritanceDepth

make a limit 6 to pass, nothing we can do with it.

> FieldCount

suppress: "it it hard to follow as most cases it it configuration related fields of Checks"

> MethodCount

suppress: "it it hard to follow as it is better to keep whole logic in one Check class"

> ClassComplexity

suppress: "it it hard to follow as it is better to keep whole logic in one Check class, we do follow cyclomatic complexity for methods"

> ClassCoupling

suppress: "we control this by checkstyle/pmd that have better suppression options"

> Deprecation DeprecatedIsStillUsed

try to fix.
cases `AbstractTypeAwareCheck` should be suppressed by javadoc annotation

> CloneableImplementsClone

activate option of it, it should fix all problems that we can inherit from other libraries.

> PublicConstructor

suppress: "we do not need to force all to have static factory methods"

> UtilityClass

suppress: "we are not pure object-oriented library, we like reasonable mix of styles, especially state-less and context-less"

> UtilityClassCanBeEnum

suppress: "we do not like this style"

> NoopMethodInAbstractClass

try to suppress on these 3 files on classes level, as we have comment that explain a reason. If not possible suppress whole rule "no options to suppress methods with comment"

>FinalClass
FinalMethod

suppress: "we are library - we do this on purpose as clear signal to users"

> FieldCanBeLocal

fix it.

> ConstantDeclaredInAbstractClass

suppress: "we are ok with constants in classes and do not like constants in interfaces"

> AnonymousInnerClass

suppress: we are ok with anonymous classes

**Comment by romani** [comment]
> Violations included:

please activate that list of inspections and send PR(not for merge) to generate report on TC.
After each merge to our master, update that PR with list of item that still unresolved, TC will keep report for you.

**Comment by Nimfadora** [comment]
@romani this task is done, for what purpose I should send PR with violations?

**Comment by romani** [comment]
ok, am closing this issue as you confirming that all are covered. 
We will recheck what is not covered at end of unmbrella issue (#4681).
